### PR TITLE
add email notification function in styles moderation

### PR DESCRIPTION
**Pull request by sumandari**
@timlinux @dimasciput 

This PR refers to #106. It does:

- send email notification to **staff** when there's a new style created.
- send email notification to **staff** when there's a style updated.
- send email notification to **creator** when the styles has been reviewed (approved/ rejected).
- include unit test for `style_notify` and `style_approval_notify` function alone.
 
the updated style will require a review from staff.

### CODE REVIEW

File: qgis-app/styles/views.py [hunk 1]
```diff
@@ -43,6 +47,92 @@ def check_styles_access(user, style):
...
+                  mail_from,
+                  recipients,
+                  fail_silently)
+
+
+def style_notify(style, created=True):
+    """
+    Email notification when a new style created.
+    """
+    recipients = [u.email for u in User.objects.filter(
```
Comment (timlinux) [inline qgis-app/styles/views.py:73]: So here I would prefer that we email specific users to let them know a new style has been added to the platform. I created the group 'Style Managers' for this purpose. Please make sure that we have documented in the README how to set up the style managers group and that the members of this group will receive style related emails.

File: qgis-app/styles/views.py [hunk 2]
```diff
@@ -43,6 +47,92 @@ def check_styles_access(user, style):
...
+
+
+def style_approval_notify(style, creator):
+    """
+    Email notification system for approval styles
+    """
+
+    recipients = [creator.email]
+
+    if settings.QGIS_DEV_MAILING_LIST_ADDRESS:
```
Comment (timlinux) [inline qgis-app/styles/views.py:108]: Again here, we should only notify the style manager group members and the individual original submitter of the style.

**Comment by timlinux** [comment]
Please also test using the https://docs.djangoproject.com/en/3.1/topics/email/#console-backend console backed and update this PR showing examples of the different emails and verifying the recipients will be correct.

**Comment by sumandari** [comment]
@timlinux 
This updated PR does decorate unit test `EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'` so that the email will be printed out in console during test, alike in [latest build](https://travis-ci.com/github/qgis/QGIS-Django/builds/203738056) line 732
The members of Style Managers group are allowed to review the styles (even if they are not staff).

The following are gif animation and console output of styles app running on  my local machine when I changed EMAIL_BACKEND value to 'django.core.mail.backends.console.EmailBackend'  in settings_docker. This is only for demo purpose. It doesn't include in this PR, and it should not be, I think.

send_email_create

console print out:
1. Create a style
```

Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: A new style has been created by user1.
From: noreply@qgis.org
To: manager1@fake.email, manager2@fake.email, manager3@fake.email
Date: Tue, 24 Nov 2020 08:36:56 -0000
Message-ID: <160620701617.754.9209710297728996935@uwsgi>


Style name is: Building 3_9nd1r
Style description is: A new style uploaded by user1
Link: http://plugins.qgis.org/styles/9/

-------------------------------------------------------------------------------
```

2. Approve style
```
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: Style Building 3 approved notification.
From: noreply@qgis.org
To: manager1@fake.email, manager2@fake.email, manager3@fake.email,
 user@fake.email
Date: Tue, 24 Nov 2020 08:37:29 -0000
Message-ID: <160620704960.754.17840928501972493770@uwsgi>


Style Building 3 approved by manager1.
Approved for testing
Link: http://plugins.qgis.org/styles/6/

-------------------------------------------------------------------------------
```

3. Reject style
```
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: Style Building 3_9nd1r rejected notification.
From: noreply@qgis.org
To: manager1@fake.email, manager2@fake.email, manager3@fake.email,
 user@fake.email
Date: Tue, 24 Nov 2020 08:37:48 -0000
Message-ID: <160620706862.754.13089011119147718488@uwsgi>


Style Building 3_9nd1r rejected by manager1.
Rejected for testing purpose
Link: http://plugins.qgis.org/styles/9/

-------------------------------------------------------------------------------
```

4. Update style
```
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: A new style has been updated by user1.
From: noreply@qgis.org
To: manager1@fake.email, manager2@fake.email, manager3@fake.email
Date: Tue, 24 Nov 2020 08:40:07 -0000
Message-ID: <160620720762.754.17931622325373376159@uwsgi>


Style name is: Building 3_9nd1r
Style description is: A new style uploaded by user1. update
Link: http://plugins.qgis.org/styles/9/

-------------------------------------------------------------------------------
```

File: qgis-app/styles/views.py [hunk 3]
```diff
@@ -26,25 +30,105 @@
...
+            recipients,
+            fail_silently=True)
+        logging.debug('Sending email notification for %s style, '
+                      'recipients:  %s' % (style.name, recipients))
+    else:
+        logging.warning('No recipients found for %s style notification'
+                        % style.name)
+
+
+def style_approval_notify(style: Style, creator: User, staff: User) -> None:
```
Comment (timlinux) [inline qgis-app/styles/views.py:97]: style_update_notify

**Comment by timlinux** [comment]
LGTM

File: qgis-app/styles/views.py (hunk 3 above)
Comment (sumandari) [reply qgis-app/styles/views.py]: the style_approval_notify method has been renamed to style_update_notify

**Pull request merged**